src/comp/comp.py

```diff
@@ -24,48 +24,50 @@ def __repr__(self):
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with 'D':
 print("Starts with D:")
-a = []
-print(a)
+a = [h.name.lower().strip("Alice Bob Charlie Eve Frank Glenn Harrison Igon").title() for h in humans]
+# a = filter(lambda h: h == 'D',humans)
 
+print(a)
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
-b = []
-print(b)
+b = [h.name.strip("Bob Frank Glenn Harrison Igon David") for h in humans]
+# b = filter(lambda h: min(h)== 'e',humans)
 
+print(b)
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 print("Starts between C and G, inclusive:")
-c = []
+c = ['C'> h.name[0] > 'G' for h in humans]
 print(c)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-d = []
+d = [h.age + 10 for h in humans]
 print(d)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-e = []
+e = [h.name + '-' +  str(h.age) for h in humans]
 print(e)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-f = []
+f = [h.age in range(27-33) for h in humans]
 print(f)
 
 # Write a list comprehension that creates a list of new Humans like the old
 # list, except with all the names uppercase and the ages with 5 added to them.
 # The "humans" list should be unmodified.
 print("All names uppercase:")
-g = []
+g = [h.name.upper() +','+ str(h.age +5) for h in humans]
 print(g)
```

Comment:
Here is an example for this one: 

"g = [Human(h.name.upper(), h.age+5) for h in humans]"

Python is very succinct. Many times you can do things with very little code. You just have to find the correct syntax.